### ISSUE TITLE
HyperParameter Tuning for BringYourOwn model

### ISSUE THREAD

**Issue by alnadaf** [awslabs/amazon-sagemaker-examples#463]
Dear all,

I have created a docker for my own regression model which is trained properly in the SageMaker. My docker structure is the same as the structure given in the example "scikit_bring_your_own.ipynb". Itried to tune the hyperparameters of my model and added the following commands to my notebook. 

estimator = sage.estimator.Estimator(image,
                       role, 1, 'ml.m4.xlarge',
                       output_path="s3://{}/output".format(sess.default_bucket()),
                       sagemaker_session=sess,
                        hyperparameters={'target': 'SS'})
from sagemaker.tuner import HyperparameterTuner, ContinuousParameter, IntegerParameter

hyperparameter_ranges = {'max_depth': IntegerParameter(1, 10),
                         'learning_rate': ContinuousParameter(0.0001, 0.001),
                        'subsample': ContinuousParameter(0, 1),
                        'max_features': ContinuousParameter(0.1, 1)}
objective_metric_name = 'mse'
metric_definitions = [{'Name': 'mse',
                       'Regex': 'mse: ([0-9\\.]+)'}]
tuner = HyperparameterTuner(estimator,
                            objective_metric_name,
                            hyperparameter_ranges,
                            metric_definitions,
                            objective_type='Minimize',
                            max_jobs=4,
                            max_parallel_jobs=2)
tuner.fit({'training': s3_input_train, 'test': s3_input_validation})

After running the above scripts, as you see in the attached image, my hyperparameter jobs are completed successfully but do not return any "Objective Metric Value". 
![scrennshot](https://user-images.githubusercontent.com/44056042/48312806-83608f80-e5cd-11e8-8f65-99220133a276.png)

I want to tune the hyperparameters of my model in SageMaker and could not find any example in this regard. I need your help to show me where is my missing part and how I can tune the hyperparameters of my own model in SageMaker.

Thank you!

**Comment by djarpin** [comment]
Hi @alnadaf - Can you confirm that there's a single line in each of your training job's CloudWatch logs of the form:
```
mse: 0.123456
```
The regex you're defining in the above code is looking for that, but doesn't seem to be capturing it.  Common mistakes might be forgetting to print that information out, or something like printing "mse=" in the training script but then defining the regex as "mse:"

Thanks.

**Comment by alnadaf** [comment]
Hi @djarpin, Thank you for your response!! I did not know that we need to print the output in the training script and no where in the AWS document mentioning that. I printed the output and it worked. Thank you again for your help!!

**Comment by alnadaf** [comment]
The issue is resolved!!
